staging/src/k8s.io/apiserver/plugin/pkg/audit/dynamic/dynamic.go

```diff
@@ -300,6 +314,11 @@ func (b *backend) updateSink(oldSink, newSink *auditregv1alpha1.AuditSink) {
 func (b *backend) deleteSink(sink *auditregv1alpha1.AuditSink) {
 	b.delegateUpdateMutex.Lock()
 	defer b.delegateUpdateMutex.Unlock()
+	if b.delegateStopped {
+		msg := fmt.Sprintf("Could not delete audit sink %q uid: %s. Update to all delegates is stopped.", sink.Name, sink.UID)
```

Comment:
Yeah, the purpose of this is just to catch some racy edge conditions while the API server is shutting down. I think we can safely ignore the errors here. Although I suppose we'd want to know if the backend was stopped erroneously... maybe just log it at warning level?

Reply:
I am so sorry guys. I thought I replied two days ago, but I just found that I didn't click commit...
About updateSink(), addSink() and deleteSink(), I really appreciate your discussion, because I am not familiar with how they will be used.
Here is what I did: For updateSink() and addSink(), I kept klog.Error(). For deleteSink(), I used klog.Warning() instead.